pkg/kapis/cluster/v1alpha1/handler.go

```diff
@@ -321,6 +322,49 @@ func (h *handler) updateKubeConfig(request *restful.Request, response *restful.R
 	response.WriteHeader(http.StatusOK)
 }
 
+func (h *handler) checkKubeConfigExpirationDate(request *restful.Request, response *restful.Response) {
+	clusterName := request.PathParameter("cluster")
+	cluster, err := h.clusterLister.Get(clusterName)
+	if err != nil {
+		api.HandleBadRequest(response, request, err)
+		return
+	}
+	config, err := loadKubeConfigFromBytes(cluster.Spec.Connection.KubeConfig)
+	if err != nil {
+		api.HandleBadRequest(response, request, err)
+		return
+	}
+	if config.CertData == nil {
+		api.HandleBadRequest(response, request, fmt.Errorf("no cert data"))
+		return
+	}
+
+	cmd := exec.Command("openssl", "x509", "-enddate", "-noout")
```

Comment:
why not use the go package `crypto/tls`? I think that's more convenient.

Reply:
I didn't know this package could do that, I'll do some research

Reply:
My bad, I mean `https://pkg.go.dev/crypto/x509`, K8s use it to generate kubeconfig, so we do in tower `https://github.com/kubesphere/tower/blob/master/pkg/certs/cert.go`.